### ISSUE TITLE
Call out that the CLI and HTTP APIs are functionally the same.

### ISSUE THREAD

**Issue by hsanjuan** [ipfs/ipfs-docs#296]
## [This issue has a bounty of $100](https://github.com/ipfs/devgrants/projects/1)

This is an issue to track curation and inclusion of https://github.com/ipfs/specs/blob/bc6a32ecff76d4b6e5a19955138305598b1a65eb/overviews/implement-api-bindings.md

That doc contains useful information on how the API works and how it relates to the CLI.

**Comment by johnnymatthews** [comment]
I don't understand the purpose of the [doc you linked to](https://github.com/ipfs/specs/blob/bc6a32ecff76d4b6e5a19955138305598b1a65eb/overviews/implement-api-bindings.md) @hsanjuan. The doc itself is pretty old (2017). Is it still relevant? Is API binding something that users currently use, or is it something they'd like to see?

**Comment by hsanjuan** [comment]
I think the actionable here is to clearly stablish that there is a 1-to-1 link between the API and the CLI commands. We should mention this in both the API and the CLI docs I guess.

We can probably live without the rest. We have bindings in most languages and I think people can follow the path of crumbs from the other docs.

**Comment by johnnymatthews** [comment]
> there is a 1-to-1 link between the API and the CLI

@hsanjuan ah right! Didn't realize that before. You're saying we should have a call-out saying _hey, the HTTP API and CLI are functionally the same_? That makes sense.

---

Further on down the line (when we've got more time), since they both act pretty much the same we could potentially have a sidebar navigation something like:

```text
APIs/
    go-ipfs
    js-ipfs
    http/cli api
        ipfs
        ipfs add
        ipfs bitswap
        ...
```

And then in each of the pages (`ipfs`, `ipfs add`, `ipfs bitswap`, etc) we could have something like:

```markdown

## ipfs add

| Description | HTTP API | Command-line |
| ----------- | -------- | ------------ |
| Add a file or directory to ipfs. | `/api/v0/add` | `ipfs add <path>...` |

```

## Tasks

But that's a discussion for another time. For now we need to:

- [ ] Add a call out to the CLI saying that it's functionally the same as the HTTP API.
- [ ] Add a call out to the HTTP API saying that it's functionally the same as the CLI.

**Comment by hsanjuan** [comment]
> hey, the HTTP API and CLI are functionally the same?

Mostly the same.

Note js-ipfs and go-ipfs API are similar but disjoint in some cases.

Your proposed solution would work, at least for go-ipfs, but we'd need to update the doc generation a bit.

**Comment by johnnymatthews** [comment]
> Mostly the same.

@hsanjuan: How are they different? Are they different enough that we should write a paragraph or page details their differences?

**Comment by hsanjuan** [comment]
I think some CLI commands have some flags which are specific to the CLI/client side of things. IIRC those are ignored during the API-docs generation, but could also not be ignored and be printed with a CLI-only warning.

**Comment by olizilla** [comment]
@johnnymatthews could you update the description on this issue to define exactly what changes are required in order to claim this bounty?

**Comment by johnnymatthews** [comment]
I've just created a new PR #518 that re-adds the `docs/reference/generate-cli-docs.sh` script. Once that's in there, @ChaituVR can you add the changes you made in #512 into `docs/reference/generate-cli-docs.sh`?

**Comment by ChaituVR** [comment]
@johnnymatthews ok will add once it is merged

**Comment by johnnymatthews** [comment]
@ChaituVR, #518 has now been merged :)
